### PR TITLE
adding surface temperature to final_data

### LINKED ISSUE #80: Include surface temperature into the final_data() function
Now that there is a script to download and process the surface temperature, it is necessary to add it to the final data function.

### PR DESCRIPTION
**Pull request by CamSalda**
Closes #80

### CODE REVIEW

File: laos_gggi/combine_data.py [hunk 1]
```diff
@@ -99,12 +109,35 @@ def final_data():
         .loc[lambda x: x.index.get_level_values(0).isin(common_codes2)]
         .copy()
     )
+    # ISO reconciliation: hadcrut
+    merged_dict_iso = merged_dict["wb_data"].index.get_level_values(0).unique()
+    hadcrut_iso = merged_dict["surface_temp"].index.get_level_values(0).unique()
+    # Codes in gpcc but not in world
+    ", ".join(list(set(hadcrut_iso) - set(merged_dict_iso)))
+    # Codes in world but not in gpcc
+    ", ".join(list(set(merged_dict_iso) - set(hadcrut_iso)))
```
Comment (jessegrabowski) [inline laos_gggi/combine_data.py:118]: Does this code do anything?

File: laos_gggi/combine_data.py [hunk 2]
```diff
@@ -99,12 +109,35 @@ def final_data():
...
         .copy()
     )
 
+    # EM-DAT damages
+    merged_dict["emdat_damage"] = (
+        merged_dict["emdat_damage"]
+        .loc[lambda x: x.index.get_level_values(0).isin(common_codes2)]
+        .copy()
+    )
+    # EM-DAT EVENTS
```
Comment (jessegrabowski) [inline laos_gggi/combine_data.py:140]: Can use a loop here:

```
for key in ['emdat_damage", "surface_temp"]:
    merged_dict[key] = (merged_dict[key].loc[...].copy()
```

File: laos_gggi/combine_data.py [hunk 3]
```diff
@@ -99,12 +109,35 @@ def final_data():
...
+    )
+
+    merged_dict["surface_temp"] = (
+        merged_dict["surface_temp"]
         .loc[lambda x: x.index.get_level_values(0).isin(common_codes2)]
         .copy()
     )
 
+    # EM-DAT damages
+    merged_dict["emdat_damage"] = (
```
Comment (jessegrabowski) [inline laos_gggi/combine_data.py:135]: why `emdat_damage` twice?

File: laos_gggi/combine_data.py [hunk 4]
```diff
@@ -175,4 +219,12 @@ def final_data():
         how="outer",
     )
 
+    merged_dict["df_time_series"] = pd.merge(
+        merged_dict["df_time_series"],
+        merged_dict["surface_temp_agg"],
+        left_index=True,
+        right_index=True,
+        how="outer",
```
Comment (jessegrabowski) [inline laos_gggi/combine_data.py:227]: this one is `outer` but the other one is `left`, is that on purpose?

**Review by jessegrabowski — APPROVED** [review]
Small comments, pre-approving so you don't to re-review. But please address at least the code duplication/unused code

File: laos_gggi/combine_data.py (hunk 1 above)
Comment (CamSalda) [reply laos_gggi/combine_data.py:118]: Actually, it does not.  Deleted.

File: laos_gggi/combine_data.py (hunk 3 above)
Comment (CamSalda) [reply laos_gggi/combine_data.py:135]: Mistake solved

**Pull request merged**